Development of the results page of parsing

ProgrammSM opened the pull request.

Commit:
Удаляем лишние комментарии в .travis.yml

Commit:
Реализуем загрузку файла, обработку и вывод результата

В SiteCоntroller доработан actionCalc(), добавлена модель CalcForm, добавлена модель для работы с таблицей в базе данных Services, переработано представление calc, добавлено представление result, добавлены файлы SQLite data.db в /data и /tests/_data/, внесены изменения в db.php и test_db.php, в test.php добавлен компонент xlsxParser, внесены изменения в web.php, добавлен юнит-тест CalcFormTest.php, внесены изменения в функциональный тест SiteCest, удалены стандартные тесты

Commit:
Вносим мелкие исправления в форматирование

artezh commented on config/test.php:
```diff
@@ -32,7 +32,10 @@
-        ],        
+        ],
+        'xlsxParser' => [
+            'class' => 'app\components\XlsxParserComponent',
```
XlsxParserComponent::class

artezh commented on controllers/SiteController.php:
```diff
@@ -22,12 +26,29 @@ public function actionIndex()
+            // Загружаем конфигурацию с базы данных.
+            $config = $model->getConfiguration();
+            /** Передаём путь на временный файл в компонент xlsxParser,
+             * конфигурацию и зарплатную ставку. */
```
*/ в следующей строке

artezh commented on controllers/SiteController.php:
```diff
@@ -22,12 +26,29 @@ public function actionIndex()
+                500.0);
+            return $this->render('result', ['model' => $model]);
+        }
+        return $this->render('calc', ['model' => $model]);
```
нужен 1 return. изменяется ведь только имя представления

artezh commented on models/CalcForm.php:
```diff
@@ -0,0 +1,43 @@
+    /**
+     * @var UploadedFile Загружаемый файл.
+     */
+    public $file;
```
зачем?

ProgrammSM replied:
Для работы UploadedFile::getInstance($model, $attribute) в SiteController во время загрузки файла во временную папку

Commit:
Оптимизируем код

Избавляемся в SiteController от лишнего return и промежуточных действий с CalcForm, удаляем с CalcForm лишнюю переменную и дополняем комментарием, исправляем представления calc и result, исправляем конфигурацию web под стандарты php 7.0

artezh commented on controllers/SiteController.php:
```diff
@@ -2,10 +2,14 @@
+ * SiteController класс для главной страницы, вычисления и отображения
+ * результатов вычисления.
  *
  * @package app\controllers
```
@package убрать

artezh commented on controllers/SiteController.php:
```diff
@@ -22,12 +26,43 @@ public function actionIndex()
+             */
+            $data
+                = [
+                'data' => Yii::$app->xlsxParser->parse($uploadedFile->tempName,
```
странное форматирование

artezh replied:
$data = [

artezh commented on models/CalcForm.php:
```diff
@@ -0,0 +1,38 @@
+     *      UploadedFile::getInstance() в SiteController, где требуется указать
+     *      модель.
+     */
+    public $file;
```
необходима валидация $file
прочитай https://github.com/yiisoft/yii2/blob/master/docs/guide-ru/input-file-upload.md

Commit:
Добавляем валидацию + Unit-тесты

Внесены изменения в SiteController, добавлены правила валидации в CalcForm, добавлены unit-тесты для проверки валидации загружаемых файлов

artezh commented on controllers/SiteController.php:
```diff
@@ -22,12 +24,44 @@ public function actionIndex()
+                 * конфигурацию и зарплатную ставку и возвращаем результат в $data.
+                 */
+                $data = [
+                    'data' => Yii::$app->xlsxParser->parse($model->file->tempName,
```
Давай перенесем работу с парсером в модель

artezh commented on models/CalcForm.php:
```diff
@@ -0,0 +1,56 @@
+                'file',
+                'extensions' => 'xlsx',
+                'checkExtensionByMimeType' => false,
+                'skipOnEmpty' => false,
```
если файл обязателен, то необходимо правило required на него

artezh commented on tests/unit/models/CalcFormTest.php:
```diff
@@ -0,0 +1,69 @@
+    public function testGetConfiguration()
+    {
+        $calcForm = new CalcForm();
+        $this->assertArraySubset([[1, 4.0], [2, 2.3], [3, 3.5]],
```
конфиг вида [['type' => 1, 'coef' => 4.0]] более читаем

Commit:
Оптимизируем код

Переносим работу с парсером из SiteController в модель CalcForm, в CalcForm создаём новый метод getData($config, $rate) для работы с компонентом xlsxParser, вносим изменения в метод getConfiguration для повышения читабельности кода, вносим изменения в XlsxParserComponent под изменения формата конфигурации, вносим изменения в unit-тесты под текущие изменения, устраняем лишний символ в представлении calc

Commit:
Исправляем комментарий в CalcForm

artezh commented on controllers/SiteController.php:
```diff
@@ -22,12 +24,40 @@ public function actionIndex()
+            $model->file = UploadedFile::getInstance($model, 'file');
+            if ($model->validate()) {
+                // Загружаем конфигурацию из базы данных.
+                $config = $model->getConfiguration();
```
config из модели достается и без изменений обратно кидается. не надо вообще доставать

artezh commented on controllers/SiteController.php:
```diff
@@ -22,12 +24,40 @@ public function actionIndex()
+                $view = 'result';
+                // Формируем данные для представления result
+                $data = [
+                    'data' => $model->getData($config, 500.0),
```
оставить один параметр, и перенести его в параметры приложения

Commit:
Оптимизируем код

artezh commented on models/CalcForm.php:
```diff
@@ -0,0 +1,69 @@
+    /**
+     * Формируем данные с помощью компонента xlsxParser
+     *
+     * @param array[] $config конфигурация для формирования файла
```
лишнее

Commit:
Удаляем лишний комментарий
